Network/Socket/Syscall.hs

```diff
@@ -200,6 +201,11 @@ accept s = withNewSocketAddress $ \sa sz -> do
      setNonBlockIfNeeded new_fd
      setCloseOnExecIfNeeded new_fd
 # endif /* HAVE_ADVANCED_SOCKET_FLAGS */
+     -- Some distros don't fill `sockaddr` appropriately in the `accept` syscall
+     -- Make an explicit call to pull the correct name
+     poke ptr_len (fromIntegral sz)
+     throwSocketErrorIfMinus1Retry_ "Network.Socket.getSocketName" $
+       c_getsockname new_fd sa ptr_len
```

Comment:
I don't understand why `c_getsockname` is used intead of `c_getpeername`.

Reply:
I believe getsockname returns information about your socket while getpeername returns information about your peer's socket. If a port number is required, getsockname will tell you your port number while getpeername will tell you your peer's port number.

I'm happy to change if you have a strong preference.

Reply:
This isn't a matter of "strong preference".  The `accept` function must and has always returned the *peer* address. The local address is fixed independently of the peer and is not particularly useful. Thus `c_getsockname` is simply wrong.

Can you elaborate on why this change is needed at all?  Of course for unix-domain sockets the peer and the local address are always the same.  If the returned sockaddr in `accept` is unreliable for unix-domain sockets on some platforms, then the right change is in the C-FFI code, to return the local address just for unix-domain sockets. Or find out what the present C-FFI code is doing wrong if there is a more correct way to get the peer address (note that unix-domain sockets need a larger sockaddr buffer than is needed for AF_INET or AF_INET6 sockets, and perhaps the provided buffer is not large enough).

Reply:
Thanks for the explanation @vdukhovni, you know much more about the nuances here than I do. If the intent of `accept` is to return the peer address then moving to `c_getpeername` makes sense.

I explained the observed behavior in the original message and included a test case with this diff that fails without the changes.
Can you be more specific what you mean by the C-FFI code? The call to the syscall is correct, but the data returned does not follow the man page (in OSX and centos at the very least). To accurately get the address data, a subsequent syscall and processing is required.
If there's a better place for this logic to go, please let me know. If you'd prefer this be guarded under certain conditions, please let me know. Making the change here let the test suite pass.